src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/EnumToIntConverter.shared.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Globalization;
+using Xamarin.CommunityToolkit.Extensions.Internals;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Converters
+{
+	/// <summary>
+	///     Converts an <see cref="Enum" /> to its underlying <see cref="int" /> value.
+	/// </summary>
+	public class EnumToIntConverter : ValueConverterExtension, IValueConverter
+	{
+		/// <summary>
+		/// Convert an <see cref="Enum"/> to corresponding underlying <see cref="int"/>
+		/// </summary>
+		/// <param name="value"><see cref="Enum"/> value to convert</param>
+		/// <param name="targetType">The type of the binding target property. This is not used.</param>
+		/// <param name="parameter">Additional parameter for converter. This is not used.</param>
+		/// <param name="culture">The culture to use in the converter. This is not used.</param>
+		/// <returns>The underlying <see cref="int"/> value of the passed value, or -1 if the value is not an enumeration type</returns>
+		public object Convert(object? value, Type? targetType, object? parameter, CultureInfo? culture) =>
+			value is Enum ? System.Convert.ToInt32(value) : -1;
```

Comment:
Can we simplify ```System.Convert``` to just ```Convert```?

Why do we return ```-1```? it can be a valid value for some enums.
Probably we should raise an exception here

Reply:
```value is Enum enumValue ? (int)enumValue : throw new ArgumentException();```

Reply:
> Can we simplify `System.Convert` to just `Convert`?

Name collision: Convert is already the name of the method, so we need to qualify it with the System namespace.

> Why do we return `-1`? it can be a valid value for some enums.
> Probably we should raise an exception here

Agreed, done

> `value is Enum enumValue ? (int)enumValue : throw new ArgumentException();`

Hardcasting to int will give a unnecessary cast warning, and is also not entirely safe for non-default enums? (i.e. https://stackoverflow.com/a/908590/5396526)